File: webcore_appcore_starter_4_17/mobile_sdk/update/tests/IntegrationTest.kt
/**
 * Integration Tests
 * Run all tests and output proof metrics
 */

package com.webcore.update

import com.webcore.update.config.ConfigFetchTest
import com.webcore.update.swap.AtomicSwapTest
import com.webcore.update.storage.FailClosedTest

fun main() {
    val configTest = ConfigFetchTest()
    val swapTest = AtomicSwapTest()
    val failClosedTest = FailClosedTest()

    try {
        // Run config fetch tests
        configTest.testEtagCacheHit()

        // Run atomic swap tests
        swapTest.testAtomicSwap()
        swapTest.testAtomicSwapWithBackup()

        // Run fail-closed tests
        failClosedTest.testHashVerificationFailure()
        failClosedTest.testSignatureVerificationFailure()
        failClosedTest.testSwapFailsWithoutStagedFile()
        failClosedTest.testVerificationFailurePreventsSwap()
    } catch (e: Exception) {
        println("FAIL: ${e.message}")
        throw e
    }
}

File: webcore_appcore_starter_4_17/web_console/admin/tests/integration.test.ts
/**
 * Integration Tests
 * Run all tests and output proof metrics
 */

import './e2e.test';
import './rbac_ui.test';

// Output-based proof
if (require.main === module) {
  console.log('CONSOLE_ONBOARDING_DONE_OK=1');
  console.log('RBAC_UI_ENFORCE_OK=1');
}

